tests/run-tests

```diff
@@ -514,13 +538,16 @@ the last matching regex is used:
     cmd_parser.add_argument('--heapsize', help='heapsize to use (use default if not specified)')
     cmd_parser.add_argument('--via-mpy', action='store_true', help='compile .py files to .mpy first')
     cmd_parser.add_argument('--keep-path', action='store_true', help='do not clear MICROPYPATH when running tests')
+    cmd_parser.add_argument('-j', '--jobs', default=1, metavar='N', type=int, help='Number of tests to run simultaneously')
+    cmd_parser.add_argument('--auto-jobs', action='store_const', dest='jobs', const=multiprocessing.cpu_count(), help='Set the -j values to the CPU (thread) count')
```

Comment:
Just an idea, but since you now added the --auto_jobs argument in most places where run-tests gets used, maybe it's more convenient to have it as default and only explicitly turn it off when needed? Or would that mess up things? Or maybe it's too 'non-standard' (at least in comparision with make or so)?

Reply:
I'm sure open to doing that, leaving `-j1` as the way to specify no parallelism.

As you may have seen elsewhere in the PR, I specify `-j1` with `--via-mpy`, because that *does* fail, probably due to a bug not safely writing the same mpy file from multiple processes simultaneously.  If we don't mind adding `-j1` explicitly in a few cases (or fix them!) then we could make `--auto-jobs` the default.

Reply:
I think `--via-mpy` is failing because those tests work by writing an intermediate `.mpy` file which has the same name for all tests.  So it could possibly be fixed by just using a unique intermediate file.